staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -41,6 +41,9 @@ type RecommendedOptions struct {
 	// admission plugin initializers to Admission.ApplyTo.
 	ExtraAdmissionInitializers func(c *server.RecommendedConfig) ([]admission.PluginInitializer, error)
 	Admission                  *AdmissionOptions
+	// ProcessInfo is used to identify events created by the server. This is currently only required if dynamic
+	// auditing is enabled.
+	ProcessInfo *ProcessInfo
 }
 
 func NewRecommendedOptions(prefix string, codec runtime.Codec) *RecommendedOptions {
```

Comment:
this doesn't look right. If you need more options for the recommended options to make sense, require them.

Reply:
so my thought was to only require it if dynamic audit was enabled, but if we want to include it up top I can do that